cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsCloudNioFileSystemProvider.scala

```diff
@@ -35,19 +34,17 @@ class DrsCloudNioFileSystemProvider(rootConfig: Config,
 
   override def getScheme: String = "dos"
 
+  val alternateScheme: String = "drs"
+
+  /*
+ * No guarantees that host or authority is present, as this isn't a true spec yet.
+ * Just need to ensure the scheme is "drs://" or "dos://", for now.
+ */
   override def getHost(uriAsString: String): String = {
-    require(uriAsString.startsWith(getScheme + "://"), s"Scheme does not match $getScheme")
-
-    /*
-     * In some cases for a URI, the host name is null. For example, for DRS urls like 'dos://dg.123/123-123-123',
-     * even though 'dg.123' is a valid host, somehow since it does not conform to URI's standards, uri.getHost returns null. In such
-     * cases, authority is used instead of host.
-     */
-    val uri = new URI(uriAsString)
-    val host = uri.getHost
-    val hostOrAuthority = if (host == null) uri.getAuthority else host
-    require(!hostOrAuthority.isEmpty, s"Bucket/Host is empty")
-
-    hostOrAuthority
+    require(uriAsString.startsWith(getScheme + "://") || uriAsString.startsWith(alternateScheme + "://"),
```

Comment:
Given that `StackOverflowException` was not caught and caused a 3 hour test run, are we sure that whatever exception comes out of this `require` is caught properly and won't cause a Cromwell freeze-up?